src/debugger/mock-ui/session.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+import readline from "readline";
+import child_process from "child_process";
+import * as DebugProtocol from "vscode-debugprotocol";
+
+//separator for messages according to the protocol
+const TWO_CRLF = "\r\n\r\n";
+
+export class Session {
```

Comment:
It is not clear to me what a Session is and why it is distinct from the adapter. Mostly, I'm bemused by the logic that deals with parsing commands and buffering.

Reply:
I moved the parsing and buffering into a separate class.